File: 007 Product Sum/Product_Sum.js
// O(n) Time | O(d) space n = elements d = depth


function productSum(array, multipler = 1){ 
	let sum = 0
	for (const element of array){
		if (Array.isArray(element)){
			sum += productSum(element, multipler + 1)
		} else {
			sum += element;
		}
	}
	return sum * multipler;
}

File: 007 Product Sum/Product_Sum.py
# O(n) Time | O(d) space n = elements d = depth


def productSum(array, multipler = 1):
	sum = 0
	for element in array:
		if type(element) is list:
			sum += productSum(element, multipler + 1)
		else:
			sum += element
	return sum * multipler